pkg/kubectl/cmd/delete.go

```diff
@@ -194,6 +203,11 @@ func (o *DeleteOptions) Complete(f cmdutil.Factory, args []string, cmd *cobra.Co
 		return err
 	}
 
+	o.DynamicClient, err = f.DynamicClient()
```

Comment:
I suggest that we remove this code and always require DynamicClient to be non-nil in `DeleteFlags.ToOptions()` method.
Alternatively, we can make all other commands (`apply`, `replace` etc) using `DeleteOptions` to also invoke this `Complete` method, and then we can remove the `dynamicClient` parameter from `DeleteFlags.ToOptions()` method...